server/app/routes/api/wines.js

```diff
@@ -0,0 +1,102 @@
+var express = require('express');
+var router = express.Router();
+var mongoose = require('mongoose')
+
+router.get('/', function(req, res) {
+	mongoose.model('Wine')
+	.find({})
+	.then(function(wines) {
+		res.json(wines)
+	})
+	.then(null, function(err) {
+		console.log(err)
+	})
+})
+
+router.get('/:id', function(req, res) {
+	var id = req.params.id
+	mongoose.model('Wine')
+	.findById(id)
+	.then(function(wine) {
+		res.json(wine)
+	})
+	.then(null, function(err) {
+		console.log(err)
+		res.sendStatus(404)
+	})
+})
+
+router.get('/:id/reviews', function(req, res) {
+	var id = req.params.id
+	mongoose.model('Wine')
+	.findById(id)
+	.then(function(wine) {
+		wine.findReviews()
+	})
+	.then(function(reviews) {
+		res.json(reviews)
+	})
+	.then(null, function(err) {
+		console.log(err)
+		res.sendStatus(404)
+	})
+})
+
+router.get('/:id/rating', function(req, res) {
+	var id = req.params.id
+	mongoose.model('Wine')
+	.findById(id)
+	.then(function(wine) {
+		wine.findRating()
+	})
+	.then(function(rating) {
+		res.json(rating)
+	})
+	.then(null, function(err) {
+		console.log(err)
+		res.sendStatus(404)
+	})
+})
+
+router.post('/', function(req, res) {
+	mongoose.model('Wine')
+	.create(req.body)
```

Comment:
duplicate wine posts? Only admins should really be able to post wines. I'll leave a comment in there for now so we can remember to address later on, if needed.